00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/hub.yaml proposal for themes.md

```diff
@@ -0,0 +1,303 @@
+```
+
+### Theme features
+
+The main goal of adding a list of features is to help compare themes based on what they offer, before one has to try them out. Keep in mind that themes (especially large ones) have many many features, and it's not our intention to make an exhaustive list of **all** the things your theme can do. Instead, ==ideally the features you list are a way to pitch to (new) users what makes your theme unique==. For that, the idea is to parse the `feature` key to identify some of the frequently sought-after features with a controlled vocabulary and automatically add links to pre-defined categories in the hub. The controlled vocabulary is based on the feature list proposed by pseudometa and extended by argentum:
+
+| Key | Hub note | Description |
+|----|-----|------|
+| `color_schemes` | `[[Themes with additional color schemes]]` | Theme has additional color schemes
+| `tables` | `[[Themes with special table support]]` | Dedicated cssclasses for tables
+|`mobile` | `[[Themes with mobile support]]` | Does the theme provide mobile support (both phones and tablets)?
+| `publish` | `[[Themes with Obsidian Publish support]]` |Does the theme have publish support?
+|`print` | `[[Themes with PDF export support]]` | Adds CSS to customize the appearance of the PDF exports
+|`gutter` | `[[Themes with gutter features]]` |Does the theme add custom indicators or other features to the gutter?
+|`snippets` | `[[Themes with official snippets]]` | Does your theme provide additional snippets to customize parts of your theme outside of the style settings plugin?
+| `images` | `[[Themes with image handling options]]` | Does your theme add any image-handling features, e.g. zooming, galleries, aligning/wrapping around text, etc.
+|`highlights` | `[[Themes with custom highlights]]` | Additional colours or ways to achieve the core `==highlight==` behaviour
+|`asides` | `[[Themes with side notes]]` | Any features related to the `<aside>` tag or any of its css workarounds, sometimes also called side notes or margin notes
+|`checklists` | `[[Themes with custom checklists]]` |Particularly if you added custom types
+| `callouts` | `[[Themes with custom callouts]]` | Does the theme add special callouts?
+| `css_classes` | `[[Themes with custom css classes]]` | Does the theme add css classes that are not covered on other features? 
+| `miscellaneous` | Only for certain keywords (see below)| Any other feature you'd like to mention as a markdown formatted string. Mentioning the following two will add links to the corresponding pages:
+
+Recognised words in the `miscellaneous` features:
+
+- `opinionated` - Will link to `[[Opinionated themes]]`. Whether the theme is opinionated (+ can you opt out of the theme's opinionatedness)
+- `minimalistic` -  Will link to [Minimalistic themes](https://publish.obsidian.md/hub/02+-+Community+Expansions/02.02+Themes+by+Category/Minimalistic+Themes) 
+
+As an example:
+
+```yaml
+theme:
+    features: 
```

Comment:
This is being rendered weirdly in Obsidian, I think there may be a mix of tabs and spaces, which is stuffing up the indentation.

Reply:
Fixed this on the latest commit (I hope... indentation is weird since yaml likes spaces, but the Obsidian outliner plugin prefers tabs).